common/autoware_test_utils/include/autoware_test_utils/autoware_test_utils.hpp

```diff
@@ -512,6 +513,62 @@ void publishToTargetNode(
   autoware::test_utils::spinSomeNodes(test_node, target_node, repeat_count);
 }
 
+/**
+ * @brief Manages publishing and subscribing to ROS topics for testing Autoware.
+ *
+ * The AutowareTestManager class provides utility functions to facilitate
+ * the publishing of messages to specified topics and the setting up of
+ * subscribers to listen for messages on specified topics. This class
+ * simplifies the setup of test environments in Autoware.
+ */
+class AutowareTestManager
+{
+public:
+  AutowareTestManager()
+  {
+    test_node_ = std::make_shared<rclcpp::Node>("autoware_test_manager_node");
+  }
+
+  template <typename MessageType>
+  void test_pub_msg(
+    rclcpp::Node::SharedPtr target_node, const std::string & topic_name, MessageType & msg)
+  {
+    if (publishers_.find(topic_name) == publishers_.end()) {
+      auto publisher = test_node_->create_publisher<MessageType>(topic_name, 10);
+      publishers_[topic_name] = std::static_pointer_cast<void>(publisher);
+    }
+
+    auto publisher =
+      std::static_pointer_cast<rclcpp::Publisher<MessageType>>(publishers_[topic_name]);
+
+    autoware::test_utils::publishToTargetNode(test_node_, target_node, topic_name, publisher, msg);
+    RCLCPP_INFO(test_node_->get_logger(), "Published message on topic '%s'", topic_name.c_str());
+  }
+
+  template <typename MessageType>
+  void set_subscriber(
+    const std::string & topic_name,
+    std::function<void(const typename MessageType::ConstSharedPtr)> callback)
+  {
+    if (subscribers_.find(topic_name) == subscribers_.end()) {
+      std::shared_ptr<rclcpp::Subscription<MessageType>> subscriber;
+      autoware::test_utils::createSubscription<MessageType>(
+        test_node_, topic_name, callback, subscriber);
+      subscribers_[topic_name] = std::static_pointer_cast<void>(subscriber);
+    } else {
+      RCLCPP_WARN(test_node_->get_logger(), "Subscriber %s already set.", topic_name.c_str());
+    }
+  }
+
+protected:
+  // Publisher
+  std::unordered_map<std::string, std::shared_ptr<void>> publishers_;
```

Comment:
1. Please use `rclcpp::PublisherBase` as the base

Reply:
Fixed: https://github.com/autowarefoundation/autoware.universe/pull/7597/commits/ed0cee986b7ed8fe3701d68f946358fda19764e7

Reply:
Fixed: https://github.com/autowarefoundation/autoware.universe/pull/7597/commits/4e9c1cadd093f6cdc1333b1d1e707271c08c5b16